### PR TITLE
Fix easy issues reported by golangci-lint

### PR DESCRIPTION
**Pull request by krnowak**
In order to have golangci-linter running over the code, I have to put the code in the `bpf` directory into a separate go module (further explanation in the commit).

I'm adding a tools submodule, so the main module does not import the golangci-lint dependencies. It isn't used yet in the build system, but can be built with `make tools/golangci-lint` and then run in the main directory with `./tools/golangci-lint run --fix`.

The fixes in this PR are of a simple kind. There are more to be done, but they need some questions to be answered.

### CODE REVIEW

File: pkg/straceback/syscall_table.go [hunk 1]
```diff
@@ -2,340 +2,340 @@
...
-  syscallNames[326] = "copy_file_range"
-  syscallNames[327] = "preadv2"
-  syscallNames[328] = "pwritev2"
-  syscallNames[329] = "pkey_mprotect"
-  syscallNames[330] = "pkey_alloc"
-  syscallNames[331] = "pkey_free"
-  syscallNames[332] = "statx"
-  syscallNames[333] = "io_pgetevents"
-  syscallNames[334] = "rseq"
+	syscallNames = make(map[int]string)
```
Comment (alban) [inline pkg/straceback/syscall_table.go:5]: `scripts/get_syscall_table.sh` needs to be fixed too.

File: pkg/straceback/syscall_table.go [hunk 2]
```diff
@@ -2,340 +2,340 @@
...
+	syscallNames[325] = "mlock2"
+	syscallNames[326] = "copy_file_range"
+	syscallNames[327] = "preadv2"
+	syscallNames[328] = "pwritev2"
+	syscallNames[329] = "pkey_mprotect"
+	syscallNames[330] = "pkey_alloc"
+	syscallNames[331] = "pkey_free"
+	syscallNames[332] = "statx"
+	syscallNames[333] = "io_pgetevents"
+	syscallNames[334] = "rseq"
```
Comment (alban) [inline pkg/straceback/syscall_table.go:340]: On my system with `kernel-headers-5.5.5-200.fc31.x86_64`, the list in `/usr/include/asm/unistd_64.h` continues until:
```
#define __NR_clone3 435
```
Maybe you can run `scripts/get_syscall_table.sh` at the same time.

File: pkg/straceback/syscall_table.go (hunk 1 above)
Comment (krnowak) [reply pkg/straceback/syscall_table.go:5]: Hm, will do it in a follow up PR. This is probably something that should be handled by the build system. Maybe through `go generate`.

File: pkg/straceback/syscall_table.go (hunk 2 above)
Comment (krnowak) [reply pkg/straceback/syscall_table.go:340]: Let's do it in the follow up PR too.

**Pull request merged**